### PR TITLE
fix: make inspect for ObjectId work

### PR DESCRIPTION
**Pull request by addaleax**
The previous `Object.defineProperty()` calls did not work because
they did not specify a property descriptor, and even if they had
worked, the signatures of `.inspect()` and `.toString()` would
have mismatched, and even if they weren’t, the result would not
have been very useful as debugging output, because it only returned
the raw hex content of the `ObjectId` and not even e.g. the fact
that this object is an `ObjectId`.

I’m happy with any other solution, but this particular one would
make my life a tiny bit easier :)

## Description

Make the `inspect` and `[util.inspect.custom]` methods on `ObjectId` work.

---

Not sure if I should open a JIRA ticket for this, but happy to do so if it helps :)

### CODE REVIEW

**Review by mbroadst — APPROVED** [review]

**Comment by mbroadst** [comment]
@addaleax thanks! Yes, can you please open a JIRA ticket a well? That will help us make better release notes

**Comment by addaleax** [comment]
@mbroadst Neal already did that for me :) https://jira.mongodb.org/browse/NODE-2875 

I wasn’t sure whether that was necessary/helpful but I’ll open tickets for everything in the future :+1:

**Review by mbroadst — APPROVED** [review]
LGTM

**Comment by addaleax** [comment]
@mbroadst Not sure if your re-approval is a hint for me to merge this, because if so, I have to tell you that I can’t :)

**Comment by mbroadst** [comment]
@addaleax thanks for the bump. I actually want to make sure we have at least two approvals from the team before changes are merged, we'll take care of merging for you

**Review by emadum — APPROVED** [review]
LGTM 👍

**Pull request merged**
